sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/internal/SealedEnumFileSpecTests.kt

```diff
@@ -352,10 +761,254 @@ class SealedEnumFileSpecTests {
+                }
+            
+                override fun enumToSealedObject(enum: TestSealedClassLevelOrderEnum): TestSealedClass<*> = when
+                        (enum) {
+                    TestSealedClassLevelOrderEnum.TestSealedClass_FirstObject -> TestSealedClass.FirstObject
+                }
+            }
+            
+            /**
+             * An isomorphic enum for the sealed class [com.livefront.TestSealedClass]
+             */
+            enum class TestSealedClassPreOrderEnum(
+                sealedObject: TestSealedClass<*>
+            ) : TestA by sealedObject {
+                TestSealedClass_FirstObject(com.livefront.TestSealedClass.FirstObject)
+            }
+            
+            /**
+             * An implementation of [com.livefront.sealedenum.SealedEnum] for the sealed class
+             * [com.livefront.TestSealedClass]
+             */
+            object TestSealedClassPreOrderSealedEnum : SealedEnum<TestSealedClass<*>>,
+                    SealedEnumWithEnumProvider<TestSealedClass<*>, TestSealedClassPreOrderEnum>,
+                    EnumForSealedEnumProvider<TestSealedClass<*>, TestSealedClassPreOrderEnum> {
+                override val values: List<TestSealedClass<*>> = listOf(
+                    TestSealedClass.FirstObject
+                )
+            
+            
+                override val enumClass: Class<TestSealedClassPreOrderEnum>
+                    get() = TestSealedClassPreOrderEnum::class.java
+            
+                override fun ordinalOf(obj: TestSealedClass<*>): Int = when (obj) {
+                    TestSealedClass.FirstObject -> 0
+                }
+            
+                override fun nameOf(obj: TestSealedClass<*>): String = when (obj) {
+                    TestSealedClass.FirstObject -> "TestSealedClass_FirstObject"
+                }
+            
+                override fun valueOf(name: String): TestSealedClass<*> = when (name) {
+                    "TestSealedClass_FirstObject" -> TestSealedClass.FirstObject
+                    else -> throw IllegalArgumentException(${qqq}No sealed enum constant ${d}name${qqq})
+                }
+            
+                override fun sealedObjectToEnum(obj: TestSealedClass<*>): TestSealedClassPreOrderEnum = when
+                        (obj) {
+                    TestSealedClass.FirstObject -> TestSealedClassPreOrderEnum.TestSealedClass_FirstObject
+                }
+            
+                override fun enumToSealedObject(enum: TestSealedClassPreOrderEnum): TestSealedClass<*> = when
+                        (enum) {
+                    TestSealedClassPreOrderEnum.TestSealedClass_FirstObject -> TestSealedClass.FirstObject
+                }
             }
 
         """.trimIndent()
 
-        assertEquals(expected, fileSpec.create().toString())
+        assertEquals(expected, fileSpec.build().toString())
     }
 }
```

Comment:
These are some badass tests.

Just to throw this out there: [here's a little library](https://github.com/tschuchortdev/kotlin-compile-testing) intended to help testing annotation processors and compiler plugins.  If you glance through the readme and it looks useful, it might be worth a try at some point.

Reply:
I saw that a little while ago myself! I definitely would like to remove some of the mocking I'm doing right now in the `SealedEnumProcessor`. Also seeing if this project can be converted to [KSP](https://github.com/android/kotlin/tree/ksp/libraries/tools/kotlin-symbol-processing-api) is on my radar, and that library also looks like they will be supporting that as well.